API/Common.pm

```diff
@@ -91,9 +91,8 @@ sub filterPlayables {
 
 	return $items if $prefs->get('playSamples');
 
-	my $t = time;
 	return [ grep {
-		($_->{released_at} ? $_->{released_at} <= $t : 1) && $_->{streamable};
```

Comment:
This would let slip tracks which a release date in the future. Does that not exist?

Reply:
No. They don't exist. This function is actually called for both` track` and `album` objects. However, as I discovered previously, the `released_at `attribute only exists for albums. The effect of the old code was to filter out unstreamable tracks and albums, as well as albums that were not yet released, unless the user had selected the prefs option to play sample tracks (`playSamples`) in which case nothing was filtered out (not sure how much sense that made actually).  This change results in only unstreamable albums being filtered out at this level, allowing all users to see the contents of as-yet-unreleased albums containing streamable tracks and play those tracks. Sample tracks, of course, will only be playable if the `playSamples` option was chosen. I guess a simpler way of saying this is that users who simply don't want to play 30-second mp3 sample tracks will no longer be prevented from seeing upcoming album releases that contain streamable pre-release tracks. (Well that was at least a shorter way, if not simpler).  😊